pkg/actors/actors.go

```diff
@@ -296,12 +296,24 @@ func (a *actorsRuntime) callRemoteActorWithRetry(
 	return nil, errors.Errorf("failed to invoke target %s after %v retries", targetAddress, numRetries)
 }
 
+func (a *actorsRuntime) getOrCreateActor(actorType, actorID string) *actor {
+	key := a.constructCompositeKey(actorType, actorID)
+
+	// This avoids allocating multiple actor allocations by calling newActor
+	// whenever actor is invoked. When storing actor key first, there is a chance to
+	// call newActor, but this is trivial.
+	val, ok := a.actorsTable.Load(key)
```

Comment:
Why call Load before LoadOrStore here? at first glance, this looks like a redundant call?

Reply:
this change is to avoid the allocation of actor instance by `newActor` whenever actor is invoked. 

For example, the below code is the current code. So whenever actor is invoked, `newActor` will be called and allocate actor instance. To avoid this, trying to load actor key first will prevent from unnecessary same actor mem allocations. 

        val, _ := a.actorsTable.LoadOrStore(key, `newActor(actorTypeID.GetActorType(), actorTypeID.GetActorId())`)
	act := val.(*actor)

Reply:
So this change will allocate actor using newActor only when actor is activated.

Reply:
Also, no harm for performance because LoadOrStore will not be called if actor is activated.